tests/h/search/query_test.py

```diff
@@ -108,6 +109,24 @@ def search(self, pyramid_request):
 
 
 class TestUserFilter(object):
+    def test_filters_annotations_by_user(self, search, Annotation):
+        Annotation(userid="acct:foo@auth2", shared=True)
```

Comment:
Might be worth adding a test case to check for searching against a particular authority. You can search for `{username}` or `{username}@{authority}`. In the latter case that should exclude annotations from other authorities.